RELEASE.md

```diff
@@ -42,6 +42,35 @@ the version for every single beta/rc release.
 BUILD_VERSION shouln't be used to override the version for actual
 releases (1.0, 1.0.1, 1.1.0 and so on).
 
+The release process can be found below.
+
+1. Check out the right branch and the right commit. This is necessary
+when not releasing from the HEAD of master.
+
+2. Tag the right commit and push it to GitHub. This is mandatory if the
+release isn't made from the HEAD of master.
+	```
+	git tag 1.0.1 3aba546aea1235
+	git push origin 1.0.1
+	```
+
+3. Write down the BUILD_VERSION or update version/CURRENT_VERSION. This
```

Comment:
version/CURRENT_VERSION is only used to auto-generate nightly releases, right? For a specific product release, we will only use BUILD_VERSION and not rely on version/CURRENT_VERSION? Please clarify this part in RELEASE.md.